modules/runtime/nuxeo-runtime-launcher/src/main/java/org/nuxeo/log4j2/MaskSensitiveDataRewritePolicy.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+
+package org.nuxeo.log4j2;
+
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.core.Core;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.rewrite.RewritePolicy;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.impl.Log4jLogEvent;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.SimpleMessage;
+
+/**
+ * @since 11.5
+ */
+@Plugin(name = "MaskSensitiveData", category = Core.CATEGORY_NAME, elementType = "rewritePolicy")
+public final class MaskSensitiveDataRewritePolicy implements RewritePolicy {
+
+    protected static final Pattern AWS_KEY_PATTERN = Pattern.compile(
+            "(A3T[A-Z0-9]|AKIA|AGPA|AIDA|AROA|AIPA|ANPA|ANVA|ASIA)([A-Z0-9]{3})[A-Z0-9]{9}([A-Z0-9]{4})");
+
+    // cards can only start with 4, 5, 6, 34, 35 or 37
```

Comment:
There's more prefixes than that. And the length is not fixed at 16. See for instance https://baymard.com/checkout-usability/credit-card-patterns. We can check the main ones but the only reliable test is the Luhn check digit.

Reply:
I made an extended Regex to take into account more Credit Card + tests.